### PR TITLE
Follow spec file sourcenames semantics in link files

### LINKED ISSUE #504: Follow specfile source name semantics in link files
The new link parsing logic in spec.py uses the resource names given in the link file verbatim.   If the link is overriding a source already defined in the spec file, the name in the link must match the name in the spec exactly otherwise the link -defined source will be added as an extra resource.

The spec file format allows the same source to be named in several different ways.  We should implement the same logic in link:

   * source, source0, source0000 are all the same
  * source 1, source001 are the same
 * patch, patch0, patch000 are the same
* ...

We should parse the integer at the end of the source name and use that as the index in the resoources dict in the spec class.   If the integer is missing, assume it is source 0.

### PR DESCRIPTION
**Pull request by mseri**
This should fix #504

### CODE REVIEW

File: planex/spec.py [hunk 1]
```diff
@@ -256,6 +258,29 @@ def series(self):
             return queue.series()
 
 
+def _parse_name(name):
+    """
+    Parse [name] a string composed by a workd and an optional number into the
```
Comment (euanh) [inline planex/spec.py]: workd -> word

File: planex/spec.py [hunk 2]
```diff
@@ -256,6 +258,29 @@ def series(self):
...
+    Raises KeyError if [name] cannot be parsed
+    """
+    matcher = re.compile(r"\A[a-zA-Z]+(\d*)\Z")
+    match = matcher.match(name)
+
+    if match is None:
+        raise KeyError(name)
+
+    (idx, ) = match.groups()
+    if idx == 0:
```
Comment (euanh) [inline planex/spec.py]: Shouldn't this be `if idx == ''`?

File: planex/spec.py (hunk 1 above)
Comment (srowe) [inline planex/spec.py]: "word"?

File: tests/test_spec.py [hunk 3]
```diff
@@ -98,84 +98,84 @@ def test_resources(self):
...
                      "SOURCES/"),
-             Patchqueue(self.spec, "patchqueue0", "http://foo/patchqueue.tar",
+             Patchqueue(self.spec, "http://foo/patchqueue.tar",
                         "link1", "PATCHES/")])
 
     def test_resource(self):
         """URLs for individual resources are correct"""
         self.assertEqual(
             self.spec.resource("path/to/ocaml-cohttp-0.9.8.tar.gz"),
-            Blob(self.spec, "source0",
```
Comment (euanh) [inline tests/test_spec.py]: Could you please make one of these tests add 'source' with no digit, to test the special case in `_parse_name`

File: planex/spec.py (hunk 2 above)
Comment (mseri) [reply planex/spec.py]: Yes, broken when rebasing, well spotted!

**Pull request merged**